confgenerator/prometheus.go

```diff
@@ -225,6 +228,11 @@ func validatePrometheus(promConfig promconfig.Config) (string, error) {
 	}
 
 	for _, sc := range promConfig.ScrapeConfigs {
+		if sc.ScrapeInterval < minScrapeInterval {
+			sc.ScrapeInterval = minScrapeInterval
+			sc.ScrapeTimeout = minScrapeInterval
+			log.Printf("scrape_interval must be at least %v; adjusting to minimum accepted value\n", minScrapeInterval)
```

Comment:
I think we should also log the fact that we're adjusting `scrape_timeout`, though admittedly I don't have a great idea in mind for the wording. Maybe we simply emit two separate logs:

```suggestion
			log.Printf("scrape_interval must be at least %v; adjusting to minimum accepted value\n", minScrapeInterval)
			log.Printf("adjusting scrape_timeout to %v\n", minScrapeInterval)
```